drivers/sensor/fxos8700/fxos8700.c

```diff
@@ -388,8 +511,14 @@ static int fxos8700_init(const struct device *dev)
 	struct fxos8700_data *data = dev->data;
 	struct sensor_value odr = {.val1 = 6, .val2 = 250000};
 
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
```

Comment:
This logic isn't quite right. We need to test if *this* device instance is on an I2C bus, not if *any* device instance is on an I2C bus.

Reply:
Acknowledged

Reply:
@MaureenHelm .. Acknowledged

Reply:
Changed as per your suggestion, please check now